packages/slice-machine/lib/builders/SliceBuilder/FieldZones/index.tsx

```diff
@@ -111,32 +107,19 @@ const FieldZones: FC = () => {
       label: string;
       widgetTypeName: FieldType;
     }) => {
-      // @ts-expect-error TS(7053) FIXME: Element implicitly has an 'any' type because expre... Remove this comment to see the full error message
-      // eslint-disable-next-line @typescript-eslint/ban-ts-comment
-      // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
-      const widget = Widgets[widgetTypeName];
-      // eslint-disable-next-line @typescript-eslint/strict-boolean-expressions
+      const widget = sliceBuilderWidgetsArray.find(
+        (sliceBuilderWidget) =>
+          sliceBuilderWidget.CUSTOM_NAME === widgetTypeName ||
```

Comment:
This came from supporting "Repeatable Group" and "Group." Since they are both group fields, they require a custom identifier. `CUSTOM_NAME` is already used for "Link to media," so it followed the convention.

This version is more accurate than one without the `CUSTOM_NAME` check since it's possible a "synthetic" field type (e.g. "Link to media") requires a specific `widget`. Let me know if I misunderstood the purpose of `widget` though!

Reply:
The `FieldType` parameter type is incorrect. `widgetTypeName` can also be Slice Machine-specific types like `"ContentRelationship"` and `"LinkToMedia"`.

I updated the type here: https://github.com/prismicio/slice-machine/pull/1335/commits/6f4c723c8ef4916a69b65bbe5cda862f05239727